### ISSUE TITLE
isSelectable() and ignored toggleActivation

### ISSUE THREAD

**Issue by pavelkryl** [davideas/FlexibleAdapter#416]
There is a method FlexibleAdapter.isSelectable() with default implementation delegating the call to an item. This is OK.

This method is also used in the body of FlexibleViewHolder.toggleActivation() where if the result is false, the view is NOT activated (still correct). However, this ignored activation request is not propagated to upper layers which matters for example in FlexibleViewHolder.onLongClick which has a return value true/false and even in this case of ignored activation returns 'true' (line 143).

I suggest that toggleActivation() has a boolean return value and consider this return value at all places where invoked.

Maybe I am missing something. What I try to achieve is that only certain items in the list can trigger multi-select mode activation. For these items I am returning FlexibleAdapter.isSelectable() = false. The item does not get selected (the multi-select does not get activated). However still the phone vibrates as if the multi select gets activated. I am now using this workaround in my ViewHolder:

        public boolean onLongClick(View view) {
            // hack to workaround bug in FlexibleAdapter - selectable = false
            FlexibleAdapter.OnItemLongClickListener oldL = mAdapter.mItemLongClickListener;
            mAdapter.mItemLongClickListener = null;
            super.onLongClick(view);
            mAdapter.mItemLongClickListener = oldL;
            // ignore
            return false;
        }

This does not make the phone vibrate, the long click gets properly ignored.

**Comment by davideas** [comment]
@pavelkryl thanks for the explanation!
If I understood well, there's haptic feedback still going on when item is NOT selectable because the long click implementation returns always true.
Indeed by looking at the code, the long click is always executed but the check is done too late in `toggleActivation()`.

I propose to fix this by checking if item is selectable, before calling the user implementation. This should resolve also the execution of that code that indeed should not occur. If not selectable the `onLongClick()` will return false.

However, I'm not sure that by returning false is enough to not vibrate the phone.
The object `View` has however options for haptic feedback, please check the Android SDK.

**Comment by davideas** [comment]
As I suspected, returning false is not ok. the haptic feedback is still there. So, there are 2 ways: remove the listener as you do OR you disable the haptic with the setter when item is unselectable. The latter is better but the client can do it himself, in its LongClick implementation in the method override (but also in the binding method should be fine).

As it is now, there is its own equilibrium between the click, long click, drag and activation events. I tried several solutions, also the one you suggested, but it got worst.

Watch out that, by returning false, the click event is also triggered.
